### PR TITLE
refactor: remove all ResponseNoMessage funcs in V2

### PR DESCRIPTION
**Pull request by cloudxxx8**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-examples/blob/master/.github/CONTRIBUTING.md.

## Issue Number:
Fix https://github.com/edgexfoundry/go-mod-core-contracts/issues/342

## What is the new behavior?
In 15-OCT-2020 Core WG meeting, we decided to remove all the xxResponseNoMessage helper functions to reduce the future development and unit test effort.
Users could call NewXXResponse with empty string in the message
peremeter instead.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: v2/dtos/responses/event_test.go [hunk 1]
```diff
@@ -80,27 +56,3 @@ func TestNewMultiEventsResponse(t *testing.T) {
...
-	expectedStatusCode := 200
-	expectedMessage := "unit test message"
-	expectedChecksum := "04698a6f20feecb8bbf7cd01e59d31ba1ce17b24ba14b71a8fb370065d951f57"
-	actual := NewUpdateEventPushedByChecksumResponse(expectedRequestId, expectedMessage, expectedStatusCode, expectedChecksum)
-
-	assert.Equal(t, expectedRequestId, actual.RequestId)
-	assert.Equal(t, expectedStatusCode, actual.StatusCode)
-	assert.Equal(t, expectedMessage, actual.Message)
-	assert.Equal(t, expectedChecksum, actual.Checksum)
-}
```
Comment (lenny-intel) [inline v2/dtos/responses/event_test.go:95]: This isn't a `NoMessageResponse` test. Why removing?

I do think it will go way for another reason, which should be in a separate PR. That reason is with the new ID required and set from start we will no longer need checksum. So the EventPushedByChecksum API should also go away.

**Review by lenny-intel — CHANGES_REQUESTED** [review]

File: v2/dtos/responses/event_test.go (hunk 1 above)
Comment (cloudxxx8) [reply v2/dtos/responses/event_test.go]: it's a mistake.  I didn't have intention to remove this...
added it back.  thanks for the catching

**Comment by cloudxxx8** [comment]
rebased

**Review by lenny-intel — APPROVED** [review]
LGTM
